plan/planbuilder.go

```diff
@@ -325,6 +329,88 @@ func (b *planBuilder) buildAdmin(as *ast.AdminStmt) Plan {
 	return p
 }
 
+func detachIndexColumns(tn *ast.TableName) (indexOffset []int, columnOffset []int) {
+	tbl := tn.TableInfo
+	var indNames []string
+	if tbl.PKIsHandle {
+		for _, col := range tbl.Columns {
+			if mysql.HasPriKeyFlag(col.Flag) {
+				indNames = append(indNames, col.Name.L)
+			}
+		}
+	}
+	indices, _ := availableIndices(tn.IndexHints, tn.TableInfo)
+	for _, ind := range indices {
+		indNames = append(indNames, ind.Columns[0].Name.L)
+	}
+	for i, col := range tbl.Columns {
+		isIndexCol := false
+		for _, ind := range indNames {
+			if ind == col.Name.L {
+				isIndexCol = true
+				break
+			}
+		}
+		if isIndexCol {
+			indexOffset = append(indexOffset, i)
+		} else {
+			columnOffset = append(columnOffset, i)
+		}
+	}
+	return
+}
+
+func (b *planBuilder) buildAnalyze(as *ast.AnalyzeTableStmt) LogicalPlan {
+	p := &Analyze{
+		baseLogicalPlan: newBaseLogicalPlan(Ana, b.allocator),
+	}
+	parser := parser.New()
+	for _, tbl := range as.TableNames {
+		indOffsets, colOffsets := detachIndexColumns(tbl)
+		result := &Analyze{
+			baseLogicalPlan: newBaseLogicalPlan(Ana, b.allocator),
+			Table:           tbl,
+			IndOffsets:      indOffsets,
+			ColOffsets:      colOffsets,
+		}
+		var tableName string
+		if tbl.Schema.L == "" {
+			tableName = tbl.Name.L
+		} else {
+			tableName = tbl.Schema.L + "." + tbl.Name.L
+		}
+		if indOffsets != nil {
+			for _, offset := range indOffsets {
+				colName := tbl.TableInfo.Columns[offset].Name.L
+				sql := "select " + colName + " from " + tableName + " order by " + colName
```

Comment:
But that would mean that I should add DataSoure, Projection and Sort for it. That is hard to write. Maybe it's ugly, but is simple. If there is simple way to do it, why choose the hard way?
If you insist that it is ugly, I will change it.